pkg/products/cloudresources/reconciler.go

```diff
@@ -139,10 +138,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 				return phase, err
 			}
 
-			// remove the namespace
-			phase, err = resources.RemoveNamespace(ctx, installation, client, operatorNamespace, r.log)
-			if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
-				return phase, err
+			// the ns can be managed by hive and removing it here can cause rhoam uninstall to get stuck
+			isHiveManaged, err := addon.OperatorIsHiveManaged(ctx, client, installation)
+			if err != nil {
+				return integreatlyv1alpha1.PhaseFailed, err
+			}
+			if !isHiveManaged {
+				phase, err = resources.RemoveNamespace(ctx, installation, client, operatorNamespace, r.log)
+				if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+					return phase, err
+				}
```

Comment:
Would it not be better to remove this logic altogether. Being hive managed is the expected work flow. Its only when we install via OLM or local while this be needed. In my opinion it would be better to update the cleanup make targets to remove the namespece. WTY?

Reply:
I'm okay with this approach 👍🏻 

When deleting a RHOAM installation, I always relied on simply deleting the RHMI CR and then waiting for the reconcile to complete. If there are other people using this approach, there might be some confusion why the namespaces/resources in them are still existing after a successful RHOAM uninstall.

This means we have to update the README to make it obvious how to properly uninstall RHOAM.

Reply:
I think I'd prefer this logic as it is, the reason why is because it gives us an extra check during our nightly runs (which are often OLM/deployment). 
If for any reason there's anything left behind the ns would not be successfully deleted which would be highlighted in the pipelines. If we ignore removing the 3scale and cro ns, then we are hoping that everything related to these namespaces got correctly deleted or, we will need to provide an additional stage on our pipeline to run cleanup commands and have them successfully complete.

Reply:
This wont give as clear a signal to resources being left behind as an extra stage in the pipelines would. I do not agree with putting development only logic into the operator as it bring it further away from production configurations. That said I am willing to let this go.